File: Sorts/SelectionSort.py
"""
Brian Taylor Vann
github.com/taylor-vann

Description:
- Module to selection sort a list

Requirements:
- None

Methods:
- selection_sort
"""


def selection_sort(a):
    n = len(a)

    if n <= 1:
        return a

    for i in range(n):
        mindx = i

        for j in range(i + 1, n):
            if a[j] < a[mindx]:
                mindx = j

        if mindx != i:
            a[mindx], a[i] = a[i], a[mindx]

File: ECMAScript/Sorts/SelectionSort.js
/*
Brian Taylor Vann
github.com/taylor-vann

Description:
- Function to selection sort a list

Requirements:
- None

Methods:
- selectionSort
*/

function selectionSort(a) {
  l = a.length;

  if(a.length < 2) {
    return a
  }

  // walk through list ...
  for(var i = 0; i < l - 1; i++) {
    // establish an index
    var indx = i;

    // walk through the rest of the list ...
    for(var j = i; j < l; j++) {
      // if current is less than index, make index equal to current ...
      if(a[indx] > a[j]) {
        indx = j;
      }

      // if index is different than start, swap values ...
      if(indx !== i) {
        var tmp = a[i];
        a[i] = a[indx];
        a[indx] = tmp;
      }
    }
  }

  return a;
}
